src/services/mongoose/merchants.ts

```diff
@@ -0,0 +1,166 @@
+import {
+  CouldNotFindMerchantFromIdError,
+  CouldNotFindMerchantFromUsernameError,
+} from "@domain/errors"
+
+import { Merchant } from "./schema"
+import { caseInsensitiveRegex, parseRepositoryError } from "./utils"
+
+interface IMerchantRepository {
+  listForMap(): Promise<BusinessMapMarker[] | RepositoryError>
+  listPendingApproval(): Promise<BusinessMapMarker[] | RepositoryError>
+  findById(id: MerchantId): Promise<BusinessMapMarker | RepositoryError>
+  findByUsername(username: Username): Promise<BusinessMapMarker[] | RepositoryError>
+  create(args: {
+    username: Username
+    coordinates: Coordinates
+    title: BusinessMapTitle
+    validated: boolean
+  }): Promise<BusinessMapMarker | RepositoryError>
+  update(args: {
+    id: MerchantId
+    coordinates: Coordinates
+    title: BusinessMapTitle
+    username: Username
+    validated: boolean
+  }): Promise<BusinessMapMarker | RepositoryError>
+  remove(id: MerchantId): Promise<void | RepositoryError>
+}
+
+export const MerchantsRepository = (): IMerchantRepository => {
+  const findById = async (
+    id: MerchantId,
+  ): Promise<BusinessMapMarker | RepositoryError> => {
+    try {
+      const result = await Merchant.findOne({ id })
+      if (!result) {
+        return new CouldNotFindMerchantFromIdError(id)
```

Comment:
It's strange to me that empty queries are considered errors, but if that's how the rest of the codebase is handling it, I suppose we can follow the pattern